### PR TITLE
MSI: 704: Sources Grid doesn't display State/Province when specified custom region entered manually

### PR DESCRIPTION
**Pull request by RomaKis**
### Fixed Issues (if relevant)
1. https://github.com/magento-engcom/msi/issues/704

### CODE REVIEW

File: app/code/Magento/Inventory/Controller/Adminhtml/Source/SourceRegionDataProcessor.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+{
+    /**
+     * @var RegionFactory
+     */
+    private $regionFactory;
+
+    /**
+     * @param RegionFactory $regionFactory
+     */
+    public function __construct(
```
Comment (naydav) [inline app/code/Magento/Inventory/Controller/Adminhtml/Source/SourceRegionDataProcessor.php]: Unused arguments

File: app/code/Magento/Inventory/Controller/Adminhtml/Source/SourceRegionDataProcessor.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    /**
+     * @param array $data
+     *
+     * @return array
+     * @throws InputException
+     */
+    public function process(array $data): array
+    {
+        $regionId = $data['region_id'] ?? 0;
```
Comment (naydav) [inline app/code/Magento/Inventory/Controller/Adminhtml/Source/SourceRegionDataProcessor.php]: We can get the same result via one string
So, looks like it is overhead to have one class for this "string"

File: app/code/Magento/Inventory/Ui/DataProvider/SourceDataProvider.php [hunk 3]
```diff
@@ -109,7 +118,10 @@ public function getData()
                     ];
                 }
             }
+        } elseif ('inventory_source_listing_data_source' === $this->name) {
```
Comment (naydav) [inline app/code/Magento/Inventory/Ui/DataProvider/SourceDataProvider.php]: This functionality is needed only for grid
See example `\Magento\InventoryCatalog\Ui\Component\Listing\Column\SalableQuantity`

File: app/code/Magento/Inventory/view/adminhtml/ui_component/inventory_source_listing.xml [hunk 4]
```diff
@@ -186,15 +186,6 @@
                 <label translate="true">Country</label>
             </settings>
         </column>
-        <column name="region_id" component="Magento_Ui/js/grid/columns/select" sortOrder="90">
```
Comment (naydav) [inline app/code/Magento/Inventory/view/adminhtml/ui_component/inventory_source_listing.xml:189]: Need to check InlineEdit in Grid

**Review by naydav — CHANGES_REQUESTED** [review]

File: app/code/Magento/Inventory/Controller/Adminhtml/Source/SourceRegionDataProcessor.php (hunk 1 above)
Comment (RomaKis) [reply app/code/Magento/Inventory/Controller/Adminhtml/Source/SourceRegionDataProcessor.php]: Fixed.

File: app/code/Magento/Inventory/Controller/Adminhtml/Source/SourceRegionDataProcessor.php (hunk 2 above)
Comment (RomaKis) [reply app/code/Magento/Inventory/Controller/Adminhtml/Source/SourceRegionDataProcessor.php]: Done.

File: app/code/Magento/Inventory/Ui/DataProvider/SourceDataProvider.php (hunk 3 above)
Comment (RomaKis) [reply app/code/Magento/Inventory/Ui/DataProvider/SourceDataProvider.php]: Done.

File: app/code/Magento/Inventory/view/adminhtml/ui_component/inventory_source_listing.xml (hunk 4 above)
Comment (RomaKis) [reply app/code/Magento/Inventory/view/adminhtml/ui_component/inventory_source_listing.xml:189]: Looks like InlineEdit Grid give ability only change source name.

File: app/code/Magento/Inventory/Controller/Adminhtml/Source/SourceRegionDataProcessor.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ */
+class SourceRegionDataProcessor
+{
+    /**
+     * @param array $data
+     * @return array
+     */
+    public function process(array $data): array
+    {
+        if (isset($data['region_id']) && '' === $data['region_id']) {
```
Comment (RomaKis) [inline app/code/Magento/Inventory/Controller/Adminhtml/Source/SourceRegionDataProcessor.php]: There is case when we can get both `region` and `region_id` and in this case `region`  should be destroyed.

**Pull request merged**